src/sentry/api/endpoints/auth_index.py

```diff
@@ -35,6 +35,83 @@ class AuthIndexEndpoint(Endpoint):
 
     permission_classes = ()
 
+    @staticmethod
+    def _reauthenticate_with_sso(request, org_id):
+        """
+        If a user without a password is hitting this, it means they need to re-identify with SSO.
+        """
+        redirect = request.META.get("HTTP_REFERER", "")
+        if not is_safe_url(redirect, allowed_hosts=(request.get_host(),)):
+            redirect = None
```

Comment:
Is there a difference between `redirect = ""` and `redirect = None`?

Reply:
i'm actually not too sure, this was code i refactored into a helper method but i assume its the same. Will test it out and make it one or the other for consistency

Reply:
theres not difference it seems. if redirect is set to `""`, then `is_safe_url` will return false and redirect will be set to `None`. I'll change the default to `None` to begin with